src/chain.rs

```diff
@@ -0,0 +1,18 @@
+//! Chain trait.
+
+/// The base Chain trait, from which all representations of a chain are derived.
+pub trait Chain {
+    /// The chain ID.
+    fn chain_id(&self) -> u64;
+
+    /// The chain name.
+    fn name(&self) -> &str;
+}
```

Comment:
I played around with this and making `Network` a supertrait of this, and heavily considered const generics. Having them would be nice, however, I felt like it increases type complexity a bit—not sure if we wanna explore that avenue @prestwich / @DaniPopes